src/System.Runtime.Extensions/tests/Performance/Perf.StreamWriter.cs

```diff
@@ -0,0 +1,119 @@
+                                }
+                                writer.Flush();
+                                stream.Position = 0;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [MemberData(nameof(WriteMemberData))]
+        public void WritePartialCharArray(int writeLength)
+        {
+            char[] buffer = new string('a', writeLength + 10).ToCharArray();
+            int innerIterations = MemoryStreamSize / writeLength;
+            int outerIteration = TotalWriteCount / innerIterations;
+            using (var stream = new MemoryStream(MemoryStreamSize))
+            {
+                using (var writer = new StreamWriter(stream, new UTF8Encoding(false, true), DefaultStreamWriterBufferSize, true))
+                {
+                    foreach (var iteration in Benchmark.Iterations)
+                    {
+                        using (iteration.StartMeasurement())
+                        {
+                            for (int i = 0; i < outerIteration; i++)
+                            {
+                                for (int j = 0; j < innerIterations; j++)
+                                {
+                                    writer.Write(buffer, 10, writeLength);
+                                }
+                                writer.Flush();
+                                stream.Position = 0;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        [Benchmark]
+        [MemberData(nameof(WriteMemberData))]
+        public void WriteString(int writeLength)
+        {
+            string value = new string('a', writeLength);
+            int innerIterations = MemoryStreamSize / writeLength;
+            int outerIteration = TotalWriteCount / innerIterations;
+            using (var stream = new MemoryStream(MemoryStreamSize))
+            {
+                using (var writer = new StreamWriter(stream, new UTF8Encoding(false, true), DefaultStreamWriterBufferSize, true))
+                {
+                    foreach (var iteration in Benchmark.Iterations)
+                    {
+                        using (iteration.StartMeasurement())
+                        {
+                            for (int i = 0; i < outerIteration; i++)
+                            {
+                                for (int j = 0; j < innerIterations; j++)
+                                {
+                                    writer.Write(value);
+                                }
+                                writer.Flush();
+                                stream.Position = 0;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> WriteMemberData()
+        {
+            for (int i = 2; i <= 10; i++)
```

Comment:
@mellinoe I see that the guidance about performance test anti-patterns was added by you recently. What would you consider the good set of sizes to test here?